src/routes/github/github-router.ts

```diff
@@ -14,6 +14,23 @@ import { GithubCreateBranchRouter } from "routes/github/create-branch/github-cre
 import { GithubRepositoryRouter } from "routes/github/repository/github-repository-router";
 import { GithubBranchRouter } from "routes/github/branch/github-branch-router";
 import { jiraSymmetricJwtMiddleware } from "~/src/middleware/jira-symmetric-jwt-middleware";
+import { Errors } from "config/errors";
+import { NextFunction, Request, Response } from "express";
+
+// TODO - Once JWT is passed from Jira for create branch this midddleware is obsolete.
+const JiraHostFromQueryParamMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+	const jiraHostQuery = req.query?.jiraHost as string;
+	const jiraHostSession = req.session?.jiraHost as string;
+	const jiraHost = jiraHostQuery || jiraHostSession;
+	if (!jiraHost) {
+		req.log.warn(Errors.MISSING_JIRA_HOST);
+		res.status(400).send(Errors.MISSING_JIRA_HOST);
+		return;
+	}
+	res.locals.jiraHost = jiraHost;
```

Comment:
Isn't this a security hole that `jiraHost` can comes from query string?

Reply:
yes, that why we are urgently getting the JWT to come from JIRA
for the time being since we use a combination of this and the github token the damage is contained.

We only allow access to things that the github token can get.... but yes needs to be addressed asap and will be the top priority out of PIR

Reply:
But because we are setting `jiraHost` into `session`, malicious user can call this endpoint once, then go to any other endpoints they want with unverified jiraHost in the session. Impact will be outside of just create-branch.

Not saying we have an imminent security hole, but if we are not careful and somewhere trust that jiraHost in session cookie too much, it may compromise data.

Reply:
Definitely, so we will patch this as high priority! and put on the actual solution!!!